pkg/cli/start_jemalloc.go

```diff
@@ -19,7 +19,8 @@
 package cli
 
 // #cgo CPPFLAGS: -DJEMALLOC_NO_DEMANGLE
-// #cgo LDFLAGS: -ljemalloc
+// #cgo !darwin LDFLAGS: -ljemalloc
+// #cgo darwin LDFLAGS: -ljemalloc -u_je_zone_register
```

Comment:
PS I don't particularly care how many places you add `-u_je_zone_register`. The test is the more important bit.